[Docs] Point collections info to docs.ansible.com and remove mazer (#2020)
ansible/galaxy: docs/docsite/rst/finding/content_types.rst

===================

> Ansible 2.8 introduced support for a new packaging format for creating and sharing Ansible content. This new format is called a Collection, and it enables the combining of

Reviewer: ```suggestion
Ansible 2.8 introduced support for a new packaging format for creating and sharing Ansible content. This new format is called a Collection, and it lets you combine
```
